### PR TITLE
Convert tests to typescript

### LINKED ISSUE #194: npm test fails with "Repository ado-extension-test-nuget-remote does not exist"
**Describe the bug**

I'm trying to run the tests locally, but `npm test` fails:

```
  JFrog Artifactory Extension Tests
    1) "before all" hook


  0 passing (2s)
  1 failing

  1) JFrog Artifactory Extension Tests
       "before all" hook:
     Error: Server responded to http://localhost:8081/artifactory/api/repositories/ado-extension-test-nuget-virtual-1599733507 with status code 400:
{
  "errors" : [ {
    "status" : 400,
    "message" : "Repository ado-extension-test-nuget-remote does not exist\n"
  } ]
}
      at Response.getBody (node_modules/http-response-object/lib/index.js:41:23)
      at createRepo (testUtils.js:289:101)
      at createTestRepositories (testUtils.js:166:5)
      at Object.initTests (testUtils.js:98:5)
      at Context.before (tests.js:24:19)
```

**To Reproduce**

Clone the repo, set the `ADO_` environment variables properly and run `npm test` as described in the readme.

**Expected behavior**

I expected the tests to pass.

**Versions**
- Artifactory Azure DevOps extension version: 1.10.2 (based on the Git tag)
- Agent operating system: Ubuntu 20.04
- JFrog Artifactory version: 7.7.3

**Additional context**
I installed Artifactory like this:

```bash
docker run --name artifactory -v ${JFROG_HOME}/artifactory/var/:/var/opt/jfrog/artifactory -d -p 8081:8081 -p 8082:8082 docker.bintray.io/jfrog/artifactory-pro:latest
```

### PR DESCRIPTION
**Pull request by RobiNino**
Also, tests suite now create a nuget remote repository which fixes #194.

### CODE REVIEW

File: tests/tests.ts [hunk 1]
```diff
@@ -1,102 +1,110 @@
...
                 retVal = ex.toString();
             }
+            assert.ok(retVal !== '', 'An exception should have been caught');
             process.env.ADO_ARTIFACTORY_PASSWORD = oldPassword;
-            assert(!retVal.toString().includes('SUPER_SECRET'), 'Output contains password');
+            assert.ok(!retVal.includes('SUPER_SECRET'), 'Output contains password');
         });
 
-        runTest(
+        runAsyncTest(
```
Comment (yahavi) [inline tests/tests.ts:68]:
```suggestion
        runSyncTest(
```
This function can be sync. See comment in line 82.
If you agree with this change, you can delete runAsyncTest and rename runSyncTest => runTest

File: tests/tests.ts [hunk 2]
```diff
@@ -1,102 +1,110 @@
...
-                }).listen(8000);
-                jfrogUtils.downloadCli().then(() => {
-                    proxyServer.close();
+                    next();
+                });
+
+                jfrogUtils.downloadCli().then((): void => {
+                    tunnel.close();
                     process.env.HTTP_PROXY = '';
                     done(cliDownloadedWithProxy ? '' : new Error('CLI downloaded without using the proxy server'));
```
Comment (yahavi) [inline tests/tests.ts:85]:
```suggestion
                await jfrogUtils.downloadCli().then((): void => {
                    tunnel.close();
                    process.env.HTTP_PROXY = '';
                    assert.ok(cliDownloadedWithProxy, 'CLI downloaded without using the proxy server');
```

File: tests/.gitignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Ignore ts outputs
+tests.js.map
+tests.d.ts
```
Comment (yahavi) [inline tests/.gitignore:3]: I think we can ignore `js.map` and `d.ts` files globally in the root `.gitignore` file

**Review by yahavi — CHANGES_REQUESTED** [review]
Great work! Looks very good!

File: tests/.gitignore (hunk 3 above)
Comment (RobiNino) [reply tests/.gitignore:3]: I think it causes problems with ignoring node_modules

**Pull request merged**